internal/controllers/database/init.go

```diff
@@ -167,21 +221,36 @@ func (r *Reconciler) initializeTenant(
 		)
 		return Stop, ctrl.Result{RequeueAfter: DefaultRequeueDelay}, err
 	}
+	ydbOpts := ydb.MergeOptions(ydb.WithCredentials(creds), tlsOptions)
 
-	err = tenant.Create(ctx, database, creds, tlsOptions)
-	if err != nil {
-		r.Recorder.Event(
-			database,
-			corev1.EventTypeWarning,
-			"InitializingFailed",
-			fmt.Sprintf("Error creating tenant %s: %s", tenant.Path, err),
-		)
-		meta.SetStatusCondition(&database.Status.Conditions, metav1.Condition{
-			Type:   DatabaseInitializedCondition,
-			Status: metav1.ConditionFalse,
-			Reason: ReasonInProgress,
-		})
-		return r.updateStatus(ctx, database, DatabaseInitializationRequeueDelay)
+	if meta.IsStatusConditionTrue(database.Status.Conditions, DatabaseWaitOperationCondition) {
+		r.checkCreateTenantOperation(ctx, database, tenant, ydbOpts)
+	} else {
+		operationID, err := tenant.Create(ctx, ydb.WithCredentials(creds), tlsOptions)
+		if err != nil {
+			r.Recorder.Event(
+				database,
+				corev1.EventTypeWarning,
+				"InitializingFailed",
+				fmt.Sprintf("Error creating tenant %s: %s", tenant.Path, err),
+			)
+			return Stop, ctrl.Result{RequeueAfter: DatabaseInitializationRequeueDelay}, err
+		}
+		if len(operationID) > 0 {
+			r.Recorder.Event(
+				database,
+				corev1.EventTypeWarning,
+				"Pending",
+				fmt.Sprintf("Tenant creation operation in progress, operationID: %s", operationID),
+			)
+			meta.SetStatusCondition(&database.Status.Conditions, metav1.Condition{
```

Comment:
Type: CreateDatabaseOperationCondition
Status: metav1.ConditionUnknown
Reason: operationID
Message:  "Waiting for CMS CreateDatabase operation to be Ready"